src/microbe_stage/Microbe.Contact.cs

```diff
@@ -964,76 +1129,93 @@ private void HandleEngulfing(float delta)
             }
         }
 
-        // Check whether we should not be being engulfed anymore
-        var hostile = hostileEngulfer.Value;
-        if (hostile != null)
-        {
-            // Dead things can't engulf us
-            if (hostile.Dead)
-            {
-                hostileEngulfer.Value = null;
-                IsBeingEngulfed = false;
-            }
-        }
-        else
+        for (int i = engulfedObjects.Count - 1; i >= 0; --i)
         {
-            IsBeingEngulfed = false;
-        }
+            var engulfedObject = engulfedObjects[i];
 
-        previousEngulfMode = State == MicrobeState.Engulf;
-    }
+            var engulfable = engulfedObject.Object.Value;
 
-    private void ProcessPhysicsForEngulfing()
-    {
-        if (State != MicrobeState.Engulf)
-        {
-            // Reset the engulfing ignores and potential targets
-            foreach (var body in attemptingToEngulf)
+            var body = engulfable as RigidBody;
+            if (body == null)
+                continue;
+
+            body.Mode = ModeEnum.Static;
+
+            if (engulfable?.PhagocytosisStep == PhagocytosisPhase.Digested)
             {
-                StopEngulfingOnTarget(body);
+                engulfedObject.TargetValuesToLerp = (null, null, Vector3.One * Mathf.Epsilon);
+                StartBulkTransport(engulfedObject, 1.5f, false);
             }
 
-            attemptingToEngulf.Clear();
+            if (!engulfedObject.Interpolate)
+                continue;
 
-            return;
+            // Ingestion hasn't completed yet but player is no longer in engulfment mode, cancel ingestion.
```

Comment:
```suggestion
            // Ingestion hasn't completed yet but the cell is no longer in engulfment mode, cancel ingestion.
```
Can't other cells also exit engulf mode?

Reply:
They can, a tiny mistake here.